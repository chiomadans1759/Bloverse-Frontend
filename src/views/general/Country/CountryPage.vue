<template>
  <main id="country-page">
    <div class="main">
        <navbar :showBackArrow="showBackArrow" :showNavigations="showNavigations" />
      <div class="container">
        <div class="row d-flex justify-content-center align-items-center">
          <div class="col-sm-7 col-lg-6 pt-5">
            <h5 class="pb-1">Select Your Favourite Country</h5>
            <p class="select-t">Selected {{consumers.selectedCountries.length}} countries</p>
            <div>
              <choose-country></choose-country>
            </div>
          </div>
        </div>
        <div class="card country-footer py-2 rounded-0">
          <div class="container">
            <div class="row d-flex justify-content-center">
              <div class="col-sm-7 col-lg-6 d-flex justify-content-between">
                <button class="btn btn-link btn-sm" type="button">Skip</button>
                <button class="btn btn-sm btn-primary">Save</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</template>

<script>
import ChooseCountry from "@/components/ChooseCountry/ChooseCountry.vue";
import Navbar from "@/components/Navbar/Navbar.vue";
import { mapState } from "vuex";

export default {
  components: { ChooseCountry, Navbar },
  data() {
    return {
      showNavigations: false,
      showBackArrow: true
    };
  },
  computed: {
    ...mapState(["consumers"])
  }
};
</script>

<style lang="scss" scoped>
@import "./CountryPage";
</style>
