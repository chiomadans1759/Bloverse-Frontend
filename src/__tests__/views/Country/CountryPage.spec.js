import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Country from '@/views/general/Country/CountryPage.vue';
import countryData from '@/fixtures/country.js';

import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('ChooseCountry Component', () => {
  let actions
  let store

  beforeEach(() => {
    actions = {
      addToCountries: jest.fn()
    }
    store = new Vuex.Store({
      state: {
        consumers: {
          selectedCountries: [...countryData],
        }
      },
      actions
    })
  });

  it('dispatches "actionInput" when input event value is "input"', () => {
    const wrapper = mount(Country, {
      store,
      localVue,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });

    wrapper.find('.overlay').trigger('click');
    expect(actions.addToCountries).toHaveBeenCalled();
    expect(wrapper.element).toMatchSnapshot();
  })
});
