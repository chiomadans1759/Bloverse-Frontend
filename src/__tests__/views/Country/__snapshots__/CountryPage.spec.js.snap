// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseCountry Component dispatches "actionInput" when input event value is "input" 1`] = `
<main
  id="country-page"
>
  <div
    class="main"
  >
    <header
      id="feeds-header"
    >
      <nav
        class="navbar navbar-expand-lg navbar-light px-5"
      >
        <div
          class="container-fluid back-nav"
        >
          <a
            class="backlink mr-4"
          >
            <div
              class="btn rounded-btn-circle d-flex justify-content-center"
            >
              <i
                class="fal fa-arrow-left back-icon"
              />
            </div>
          </a>
           
          <span
            class="back-text"
          >
            Back
          </span>
           
          <a
            class="router-link"
          >
            <img
              class="nav-logo"
              src="@/assets/Bloverse.svg"
            />
             
            <!---->
          </a>
           
          <!---->
           
          <!---->
        </div>
      </nav>
       
      <main
        id="bl-modal"
      >
        <div
          aria-hidden="true"
          aria-labelledby="customModal"
          class="modal fade"
          id="feedback-modal"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="modal-dialog modal-dialog-centered"
            role="document"
          >
            <div
              class="modal-content"
            >
              <div
                class="modal-header darkBackground"
              >
                <!---->
                 
                <div>
                  <h6>
                    <strong
                      class="alt-title"
                    >
                      Send Us Some Feedback
                    </strong>
                  </h6>
                   
                  <h6
                    class="mt-2 sub-title"
                  >
                    Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!
                  </h6>
                </div>
                 
                <button
                  aria-label="Close"
                  class="close"
                  data-dismiss="modal"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    Ã—
                  </span>
                </button>
              </div>
               
              <div
                class="modal-body"
              >
                <div
                  class="bl-modal-content"
                >
                  <p>
                    Enter your feedback here!
                  </p>
                   
                  <textarea
                    class="form-control mb-2 p-1"
                  />
                   
                  <label
                    class="ml-1 mr-3"
                  >
                    <input
                      name="feedback-type"
                      type="radio"
                    />
                     Bug
      
                  </label>
                   
                  <label
                    class="mr-3"
                  >
                    <input
                      name="feedback-type"
                      type="radio"
                    />
                     Comment
      
                  </label>
                   
                  <label
                    class="mr-3"
                  >
                    <input
                      name="feedback-type"
                      type="radio"
                    />
                     Other
      
                  </label>
                </div>
              </div>
               
              <div
                class="modal-footer"
              >
                <button
                  class="btn btn-outline-primary"
                  data-dismiss="modal"
                  type="button"
                >
                  
              Close
            
                </button>
                 
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Send Feedback
                </button>
              </div>
            </div>
          </div>
        </div>
      </main>
    </header>
     
    <div
      class="container"
    >
      <div
        class="row d-flex justify-content-center align-items-center"
      >
        <div
          class="col-sm-7 col-lg-6 pt-5"
        >
          <h5
            class="pb-1"
          >
            Select Your Favourite Country
          </h5>
           
          <p
            class="select-t"
          >
            Selected 16 countries
          </p>
           
          <div>
            <div
              id="choose-country"
            >
              <ul
                class="row d-flex flex-wrap"
              >
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        South Africa
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        UAE
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        United Kingdom
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        United States
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Australia
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Canada
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        France
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Germany
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Ghana
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Hong Kong
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        India
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Ireland
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Italy
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Kenya
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Mexico
                      </p>
                    </div>
                  </div>
                </li>
                <li
                  class="col-6 col-md-4 col-lg-4 col-xl-4 mb-3"
                >
                  <div>
                    <div
                      class="card p-0"
                    >
                      <div
                        class="country-img-wrapper"
                      >
                        <span
                          class="overlay"
                          id="cat-span"
                          style="display: none;"
                        >
                          <div
                            class="circle"
                          />
                           
                          <i
                            class="fal fa-check fa-1x overlay-icon"
                          />
                        </span>
                         
                        <img
                          alt=""
                          class="countryname"
                          src="test-file-stub"
                        />
                      </div>
                       
                      <p
                        class="p-2"
                      >
                        Singapore
                      </p>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="card country-footer py-2 rounded-0"
      >
        <div
          class="container"
        >
          <div
            class="row d-flex justify-content-center"
          >
            <div
              class="col-sm-7 col-lg-6 d-flex justify-content-between"
            >
              <button
                class="btn btn-link btn-sm"
                type="button"
              >
                Skip
              </button>
               
              <button
                class="btn btn-sm btn-primary"
              >
                Save
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
